File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let sum = 0, maxx = -Infinity;
        for(let i = 0; i < nums.length; i++){
            if(sum > 0)sum += nums[i];
            else sum = nums[i];
            maxx = Math.max(sum, maxx);
        }
        return maxx;
};

File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE, temp = 0;
        for(int i : nums){
            temp += i;
            max = Math.max(temp,max);
            if(temp < 0) temp = 0;
        }
        return max;
    }
}